nilearn/image/image.py

```diff
@@ -638,6 +641,9 @@ def threshold_img(img, threshold, mask_img=None):
     img_data = _safe_get_data(img).copy()
     affine = img.get_affine()
 
+    if np.isnan(np.sum(img_data)):
```

Comment:
Why not just ``np.nan_to_num()`` on a general basis? This would only visit each element in img_data once.